src/components/hooks/useLayout.ts

```diff
@@ -76,7 +77,11 @@ export default (
         }).flat(10).filter(Boolean)
     ));
     return {
-        index: getIndex(schema.properties, editors, visibleProperties),
+        index: getIndex(
+            schema.properties,
+            editors,
+            fieldNames(Object.keys(cards), cards, schema, editors).fields
```

Comment:
Using Object.keys(cards) is not quite correct, because this is all cards, not the ones in the active layout. I will fix it.